docs/source/components/nodes.rst

```diff
@@ -3,20 +3,53 @@
+Nodes are the building blocks when populating the :ref:`Pipeline`. Each node provides a specific functionality on the DepthAI, a set of configurable
 properties and inputs/outputs. After you create a node on a pipeline, you can also configure it as desired and link it to other nodes.
 
-.. rubric:: Outputs and inputs
+On the table of contents (left side of the page) all nodes are listed under the :code:`Node` entry. You can click on them to find out more.
+
+.. rubric:: Inputs and outputs
+
+Each node can have zero, one or multiple inputs and outputs. For example :ref:`SystemLogger` node has no inputs and 1 output and :ref:`EdgeDetector` has
+2 inputs and 1 output, as shown below. :ref:`Script` node can have any number of inputs/ouputs.
+
+.. code-block::
+
+                 ┌───────────────────┐
+  inputImage     │                   │
+  ──────────────►│                   │
+                 │                   │ outputImage
+                 │   EdgeDetector    ├───────────►
+  inputConfig    │                   │
+  ──────────────►│                   │
+                 │                   │
+                 └───────────────────┘
+      EdgeDetector node has 2 inputs and 1 output
 
-Each node can have zero, one or multiple inputs and outputs. For example :ref:`SystemLogger` node has no inputs and 1 output and :ref:`StereoDepth` has
-2 inputs and 6 outputs.
 
 .. rubric:: Node input
 
 Node input queue is a queue for the :ref:`Messages`. It can be linked with other node's output (that's how you link up nodes). Node inputs are
-configurable - with :code:`input.setBlocking(bool)` and :code:`input.setQueueSize(num)`. Default behaviour is blocking and a queue size of 8 messages.
-If the input queue fills up, behavior of the input depends on blocking attribute.
-If blocking is enabled, new messages will be discarded until user gets the old messages. If blocking is disabled, new messages will push out old messages.
+configurable - with :code:`input.setBlocking(bool)` and :code:`input.setQueueSize(num)`, eg. :code:`edgeDetector.inputImage.setQueueSize(10)`.
+**Default behavior is blocking and a queue size of 8 messages**. If the input queue fills up, behavior of the input depends on blocking attribute.
+
+Let's say we have linked :ref:`ColorCamera` :code:`preview` output with :ref:`NeuralNetwork` :code:`input` input.
+
+.. code-block::
+
+  ┌─────────────┐                    ┌───────────────┐
+  │             │                    │               │
+  │             │ preview      input │               │
+  │ ColorCamera ├───────────────────►│ NeuralNetwork │
+  │             │     [ImgFrame]     │               │
+  │             │                    │               │
+  └─────────────┘                    └───────────────┘
+
+If **input is set to blocking mode**, and input queue fills up, no new messages from ColorCamera will be able to enter the input queue. This means messages
```

Comment:
Initially leave out the pool&limited number of messages behavior, and describe separately

Something along the lines of:
> This means ColorCamera will block and wait with sending its messages till it can push the message to the queue of NeuralNetwork input. If ColorCamera `preview` is connected to multiple inputs, the same behavior implies, with the messages being pushed sequentially to each input. Warning, depending on pipeline configuration, this can sometimes lead to pipeline freezing, if some blocking input isn't being consumed, etc...

 > (Describe pools..) Also most nodes have limited number of messages that can circulate the pipeline of one output at once (pools). This setting can usually be changed by a property. Beware, if some node holds too many messages in its queue, the producing Node might have to wait till those messages are again released, to be able to send more messages.

...

> If **blocking is disabled**, new messages will push out old messages. This eliminates the risk of pipeline freezing, but can result in dropped messages (eg. :ref:`ImgFrame`).

Modify "can result in losing messages" to "can result in dropped messages"